WooCommerce/Classes/ViewModels/MainTabViewModel.swift

```diff
@@ -42,39 +55,68 @@ final class MainTabViewModel {
     ///
     func startObservingOrdersCount() {
         observeBadgeRefreshNotifications()
+        updateBadgeFromCache()
         requestBadgeCount()
     }
 }
 
 
 private extension MainTabViewModel {
+
+    /// Construct `ResultsController` with `siteID`
+    ///
+    func updateStatusResultsController(siteID: Int64) -> ResultsController<StorageOrderStatus> {
+        let predicate = NSPredicate(format: "siteID == %lld AND slug == %@", siteID, OrderStatusEnum.processing.rawValue)
+        return ResultsController<StorageOrderStatus>(storageManager: ServiceLocator.storageManager, matching: predicate, sortedBy: [])
+    }
+
+    /// Connect hooks on `ResultsController` and query cached data
+    ///
+    func configureResultsController() {
```

Comment:
Another nit, probably `configureStatusResultsController` makes the code easier to read!